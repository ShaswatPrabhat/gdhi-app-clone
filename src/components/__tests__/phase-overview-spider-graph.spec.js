import { mount } from "@vue/test-utils";
import { describe, beforeEach, it, expect, vi } from "vitest";
import PhaseOverviewSpiderGraph from "../graphs/phase-overview-spider-graph/phase-overview-spider-graph.vue";
import Chart from "chart.js/auto";

describe("Phase OverView Spider Graph", () => {
  let wrapper;
  vi.mock("chart.js/auto");
  beforeEach(() => {
    wrapper = mount(PhaseOverviewSpiderGraph, {
      propsData: {
        countryDataCategories: [
          { name: "Phase1", phase: 1 },
          { name: "Phase2", phase: 2 },
          { name: "Phase3", phase: 3 },
          { name: "Phase4", phase: 4 },
        ],
        regionalDataCategories: [
          { phase: 1 },
          { phase: 1 },
          { phase: 2 },
          { phase: 2 },
        ],
        countryName: "Sample Country Name",
      },
    });
  });
  it("should render Graph when mounted and create a new Chart instance", () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
    vi.useFakeTimers();
    vi.advanceTimersByTime(700);
    expect(Chart.mock.calls.length).toEqual(1);
    expect(Chart.mock.calls[0][1]).toMatchSnapshot();
    // TODO: Fix mock of ChartJS to be able to assert watch call
    // wrapper.vm.$options.watch.countryDataCategories.call();
  });
});
