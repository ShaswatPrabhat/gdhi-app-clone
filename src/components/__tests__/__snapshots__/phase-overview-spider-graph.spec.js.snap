// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Phase OverView Spider Graph > should render Graph when mounted and create a new Chart instance 1`] = `
<div
  class="canvas-container"
  data-v-a4c9761d=""
>
  <canvas
    data-v-a4c9761d=""
    id="phase-overview-spider-graph"
  />
</div>
`;

exports[`Phase OverView Spider Graph > should render Graph when mounted and create a new Chart instance 2`] = `
{
  "data": {
    "datasets": [
      {
        "backgroundColor": "rgba(65,91,163,0.5)",
        "borderColor": "rgba(65,91,163,0.5)",
        "borderWidth": 1,
        "data": [
          1,
          2,
          3,
          4,
        ],
        "fill": true,
        "label": "Sample Country Name",
        "pointBackgroundColor": "rgba(65,91,163,0.5)",
        "pointBorderColor": "#fff",
        "pointHoverBackgroundColor": "#fff",
        "pointHoverBorderColor": "rgba(65,91,163,0.5)",
        "spanGaps": true,
      },
      {
        "backgroundColor": "rgba(108,117,125,0.5)",
        "borderColor": "rgba(108,117,125,0.5)",
        "borderWidth": 1,
        "data": [
          1,
          1,
          2,
          2,
        ],
        "fill": true,
        "label": "Global Average",
        "pointBackgroundColor": "rgba(108,117,125,0.5)",
        "pointBorderColor": "#fff",
        "pointHoverBackgroundColor": "#fff",
        "pointHoverBorderColor": "rgba(108,117,125,0.5)",
        "spanGaps": true,
      },
    ],
    "labels": [
      "Phase1",
      "Phase2",
      "Phase3",
      "Phase4",
    ],
  },
  "options": {
    "elements": {
      "line": {
        "borderWidth": 1,
      },
    },
    "layout": {
      "autoPadding": false,
      "padding": 0,
    },
    "plugins": {
      "legend": {
        "align": "start",
        "display": true,
        "labels": {
          "boxHeight": 20,
          "boxWidth": 20,
          "font": {
            "size": 12,
          },
          "padding": 10,
        },
        "position": "bottom",
        "rtl": false,
      },
      "tooltip": {
        "rtl": false,
      },
    },
    "scales": {
      "r": {
        "alignToPixels": true,
        "angleLines": {
          "display": true,
        },
        "beginAtZero": true,
        "pointLabels": {
          "backdropPadding": 0,
          "callback": [Function],
          "font": {
            "family": "'InterRegular', sans-serif",
            "size": 9,
          },
          "padding": 2,
        },
        "suggestedMin": 0,
        "ticks": {
          "format": "",
          "precision": 0,
        },
      },
    },
  },
  "type": "radar",
}
`;
